images/metering-ansible-operator/roles/meteringconfig/defaults/main.yml

```
@@ -103,7 +103,68 @@ _storage_overrides:
 
 meteringconfig_storage_overrides: "{{ _storage_overrides[meteringconfig_default_values | combine(meteringconfig_spec_overrides) | json_query('storage.hive.type')] | default({}, true) }}"
 
-meteringconfig_spec: "{{ meteringconfig_default_values | combine(meteringconfig_default_image_overrides, meteringconfig_storage_overrides, meteringconfig_spec_overrides, recursive=True) }}"
+#
+# Enable TLS by default
+#
+meteringconfig_reporting_operator_server_ca_certificate: ""
+meteringconfig_reporting_operator_client_cert: ""
+meteringconfig_reporting_operator_client_ca_certificate: ""
+meteringconfig_reporting_operator_client_key: ""
+
+meteringconfig_presto_server_secret_name: ""
+meteringconfig_presto_server_ca_certificate: ""
+meteringconfig_presto_server_certificate: ""
+meteringconfig_presto_server_key: ""
+meteringconfig_presto_tls_create_secret: false
+
+meteringconfig_presto_client_secret_name: ""
+meteringconfig_presto_client_ca_certificate: ""
+meteringconfig_presto_client_certificate: ""
+meteringconfig_presto_client_key: ""
+meteringconfig_presto_auth_create_secret: false
+
+_tls_overrides:
+  reporting-operator:
+    spec:
+      config:
+        prestoTLS:
+          enabled: true
+          createSecret: true
+          secretName: "reporting-operator-presto-tls-gen"
+          caCertificate: "{{ meteringconfig_reporting_operator_server_ca_certificate }}"
+
+        prestoAuth:
+          enabled: true
+          createSecret: true
+          secretName: "reporting-operator-presto-auth-gen"
+          certificate: "{{ meteringconfig_reporting_operator_client_cert }}"
+          key: "{{ meteringconfig_reporting_operator_client_key }}"
+          caCertificate: "{{ meteringconfig_reporting_operator_client_ca_certificate }}"
+  presto:
+    spec:
+      presto:
+        config:
+          tls:
+            certificate: "{{  meteringconfig_presto_server_certificate }}"
+            key: "{{ meteringconfig_presto_server_key }}"
+            caCertificate: "{{ meteringconfig_presto_server_ca_certificate }}"
+
+            enabled: true
+            createSecret: "{{ meteringconfig_presto_tls_create_secret }}"
+            secretName: "{{ meteringconfig_presto_server_secret_name }}"
+          auth:
+            certificate: "{{  meteringconfig_presto_client_certificate }}"
+            key: "{{ meteringconfig_presto_client_key }}"
+            caCertificate: "{{ meteringconfig_presto_client_ca_certificate }}"
+
+            enabled: true
+            createSecret: "{{ meteringconfig_presto_auth_create_secret }}"
+            secretName: "{{ meteringconfig_presto_client_secret_name }}"
+
+# what happens if CR is unset
+meteringconfig_tls_overrides: "{{ meteringconfig_default_values.tls.enabled | ternary(_tls_overrides, {}) }}"
```

I think we should probably also check `meteringconfig_spec_overrides.tls.enabled` probably, otherwise I don't think setting `spec.tls.enabled` in your MeteringConfig will have any effect. It's a bit tricky though, since the user may no have a `spec.tls` section, it's easiest to use `json_query` to get that value, and then use the `default` filter to fallback to the default as defined in `values.yaml`.

Eg:

```suggestion
meteringconfig_tls_enabled: "{{ meteringconfig_spec_overrides | json_query('tls.enabled') | default(meteringconfig_default_values.tls.enabled, true) }}"
meteringconfig_tls_overrides: "{{ meteringconfig_tls_enabled | ternary(_tls_overrides, {}) }}"
```